libs/ui/lib/tooltip/Tooltip.tsx

```diff
@@ -83,14 +86,22 @@ export const Tooltip = forwardRef(
       ref: mergeRefs([refs.setReference, elRef]),
     })
 
+    const isInModal = useIsInModal()
+    const isInSideModal = useIsInSideModal()
+    const zIndex = isInModal
+      ? 'z-modalDropdown'
+      : isInSideModal
+        ? 'z-sideModalDropdown'
+        : 'z-contentDropdown'
+
```

Comment:
Since the logic is the same and feels unlikely to diverge, might as well share it:

```ts
function usePopoverZIndex() {
  const isInModal = useIsInModal()
  const isInSideModal = useIsInSideModal()
  return isInModal
    ? 'z-modalDropdown'
    : isInSideModal
      ? 'z-sideModalDropdown'
      : 'z-contentDropdown'
}
```